File: count_pair_sum_matrix.cpp
//User function template for C++
class Solution{
public:	
	
	int countPairs(vector<vector<int>> &mat1, vector<vector<int>> &mat2, int n, int x)
	{
	    // Your code goes here
	    // Your code goes here
        set<int>s;
        int cnt = 0;
        for(int i=0;i<n;i++){
            for(int j = 0;j<n;j++){
                s.insert(mat1[i][j]);
            }
        }
         for(int i=0;i<n;i++){
            for(int j = 0;j<n;j++){
                auto it = s.find(x-mat2[i][j]);
                if(it!=s.end()){
                    cnt++;
                }
            }
        }
        return cnt;
	}
};

File: count_pair_sum_matrix.py
n=3
x=21
count = 0

mat1 = [[1,5,6],
        [8,10,11],
        [15,16,18]
        ]

mat2 = [[2,4,7],
        [9,10,12],
        [13,16,20]
        ]

l = []

for i in mat2:
     for j in i:
          l.append(j)

print(l)

for i in mat1:
     for j in i:
          now = x-j
          if now in l:
               print("(",j,now,")")
               count+=1


print(count)





